### PR TITLE
Bug 1173900 - Home tab isn't initialized after clearing private data and pressing show introduction

### PR DESCRIPTION
**Pull request by ackratos**

### CODE REVIEW

**Comment by fxios-buildasaur** [comment]
Result of integration 1
Integration took 17 seconds.
? build errors: checkout-error

**Comment by rnewman** [comment]
I tested this, and it works, but I actually don't know why! We close every tab. We set the index to 0, not to -1. At some point something opens a new about:home tab. But I don't see what does it!

@sleroux, could you take a second look before merging this?

**Comment by sleroux** [comment]
@rnewman Looks like we set the selectedIndex to -1 when we call selectTab: https://github.com/ackratos/firefox-ios/blob/showIntroAfterClearPrivate/Client/Frontend/Browser/TabManager.swift#L125. What probably is happening is tabs.count is already 0, which will cause the for loop block to not run at all which sets the value of selectedIndex.@ackratos I think the fix for this problem would be to check for this condition inside selectTab.

**Comment by ackratos** [comment]
@rnewman @sleroux Thank you for your kind recommendation. I will check the condition mentioned by @sleroux

**Comment by ackratos** [comment]
To @rnewman 
I think I have found the root cause of this problem. It because when we remove the last tab. TabManagerDelegate will add the "initial guide tab", but it doesn't select that tab. So the first fix works only when assign 0 to _selectedIndex. But we both believe that -1 is the right initial value of _selectedIndex.

To @sleroux 
When the TabTrayViewController showed, the "count" property of tabmanager is 1. It's right and because of the tab manager delegate will add a tab after remove the last tab. It was [this call](https://github.com/ackratos/firefox-ios/blob/showIntroAfterClearPrivate/Client/Frontend/Browser/TabTrayController.swift#L513) caused the collectionViewController fail.

In a nutshell, it was the lack of selectTab() after addTab() caused this bug.
